enhancements/hide-container-mountpoints.md

```diff
@@ -0,0 +1,339 @@
+- Add a CI lane that runs all current e2e tests with this feature enabled
+- Update the ZTP DU profile to use the new mechanism instead of the current
+  MachineConfig-based proof-of-concept
+
+#### Tech Preview -> GA
+
+- Enable the feature by default
+- Remove the CI lane that enables the feature, as it is enabled by default
+- User-facing documentation to detail the mechanism to inspect container mounts
+  or disable the feature if needed
+
+#### Removing a deprecated feature
+
+Not applicable.
+
+### Upgrade / Downgrade Strategy
+
+Not applicable.
+
+### Version Skew Strategy
+
+Not applicable. The mount namespace is fully-contained and isolated within each
+node of a cluster. There is no impact of having the feature enabled on some
+nodes and disabled on others.
+
+### Operational Aspects of API Extensions
+
+Not applicable; no API extensions; but there are operational impacts of this
+change, detailed in the 'Risks and Mitigations' section above.
+
+#### Failure Modes
+
+- If either the Kubelet or CRI-O services end up in different namespaces from
+  one another, containers started by CRI-O will not see mounts made by Kubelet,
+  such as secrets or configmaps.
+
+- If the namespace is not configured correctly to allow mounts from the OS to
+  be shared into Kubelet or CRI-O, system mountpoints will not be visible to
+  either Kubelet or CRI-O or runing containers.
+
+#### Support Procedures
+
+When this feature is enabled, a shell on a node and the 'oc debug' container
+will not have visibility of the Kubernetes mountpoints.
+
+- To start a shell within the container mount namespace, execute the
+  `kubensenter` script.
+
+- To disable this feature, inject a MachineConfig that disables the
+  'kubens.service':
+
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 99-custom-disable-kubens-worker
+spec:
+  config:
+    ignition:
+      version: 2.2.0
+    systemd:
+      units:
+      - enabled: false
+        name: kubens.service
+```
+
+## Implementation History
+
+This proposal differs from the original proof-of-concept by:
+- Moving the responsibility of entering the namespace to the tools that
+  run in it (CRI-O and Kubelet), instead of a fragile systemd
+  ExecStart-patching drop-in
+- Building in the simple off/on switch of enabling/disabling a single
+  systemd service, instead of having it tied to a monolithic MachineConfig
+  object.
+
+### Initial proof-of-concept
+
+Original work [here](https://github.com/lack/redhat-notes/tree/main/crio_unshare_mounts).
+
+It has MC objects that create:
+- The new `container-mount-namespace.service` service
+- Override files for both `crio.service` and `kubelet.service` which add the
+  appropriate systemd dependencies upon the `container-mount-namespace.service`
+  and wrap ExecStart inside of `nsenter`
+- A convenience utility called `/usr/local/bin/nsenterCmns` which can be used by
```

Comment:
nit: Maybe keep the utitlity name all small caps?

Reply:
This name already exists in the wild as pre-existing work; the proposed name of the similar utility will be `kubensenter`